src/overlays/actors/ovl_En_Jso2/z_en_jso2.h

```diff
@@ -2,16 +2,109 @@
 #define Z_EN_JSO2_H
 
 #include "global.h"
+#include "assets/objects/object_jso/object_jso.h"
 
 struct EnJso2;
 
+#define EN_JSO2_GET_TYPE(thisx) ((thisx)->params)
+
+#define EN_JSO2_AFTERIMAGE_COUNT 20
+#define EN_JSO2_FLAME_COUNT 6
+
 typedef void (*EnJso2ActionFunc)(struct EnJso2*, PlayState*);
 
+typedef enum {
+    // This type is exclusively used by the Garo Master in Stone Tower Temple who guards the Light Arrow.
+    // It has an intro and death cutscene, and the intro cutscene uses many hardcoded coordinates which
+    // assume the Garo Master is in the Light Arrow room.
+    /* 0 */ EN_JSO2_TYPE_LIGHT_ARROW_ROOM,
+    
+    // This type is used everywhere outside of Stone Tower Temple, and it lacks an intro or death cutscene.
+    /* 1 */ EN_JSO2_TYPE_NORMAL
+} EnJso2Type;
+
+typedef enum {
+    /*  0 */ EN_JSO2_BODYPART_LEFT_SWORD,
+    /*  1 */ EN_JSO2_BODYPART_RIGHT_SWORD,
+    /*  2 */ EN_JSO2_BODYPART_ROBE_TOP,
+    /*  3 */ EN_JSO2_BODYPART_ROBE_BACK,
+    /*  4 */ EN_JSO2_BODYPART_ROBE_LEFT,
+    /*  5 */ EN_JSO2_BODYPART_ROBE_RIGHT,
+    /*  6 */ EN_JSO2_BODYPART_ROBE_FRONT,
+    /*  7 */ EN_JSO2_BODYPART_HEAD,
+    /*  8 */ EN_JSO2_BODYPART_RIGHT_THIGH,
+    /*  9 */ EN_JSO2_BODYPART_RIGHT_FOOT,
+    /* 10 */ EN_JSO2_BODYPART_LEFT_THIGH,
+    /* 11 */ EN_JSO2_BODYPART_LEFT_FOOT,
+    /* 12 */ EN_JSO2_BODYPART_MAX
+} EnJso2BodyPart;
+
 typedef struct EnJso2 {
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x134];
+    /* 0x0144 */ SkelAnime skelAnime;
+    /* 0x0188 */ Vec3s jointTable[GARO_MASTER_LIMB_MAX];
+    /* 0x0200 */ Vec3s morphTable[GARO_MASTER_LIMB_MAX];
     /* 0x0278 */ EnJso2ActionFunc actionFunc;
-    /* 0x027C */ char unk_27C[0xE14];
+    /* 0x027C */ struct_80122744_arg1 unk_27C;
```

Comment:
Couldn't help but to be a bit curious about this. Here's a first draft of what's going on with this struct:
https://github.com/zeldaret/mm/compare/main...engineer124:OverrideInput?expand=1
It basically allows to move player to a list of points. Maybe that'll help for now while I try to polish these names

Reply:
Oh, thanks! I tried to name things appropriately here, and I also fixed the values of what is now called `sPlayerOverrideInputPosList` actually use negative numbers correctly (I forgot that Vec3s is called Vec3s because it uses s16s lol)